### PR TITLE
HTML head customisation with `head.hbs`

### PR DESCRIPTION
**Pull request by GillesRasigade**
Allow to define custom attributes within the HTML <head> tag thanks to a `head.hbs` template file.

A typical use case is the addition of mobile capabilities for PWA deployment for example to render the doc in fullscreen once added to the home screen (Android or iOS):

```html
<meta name="apple-mobile-web-app-capable" content="yes" />
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="mobile-web-app-capable" content="yes">
```

### CODE REVIEW

File: src/renderer/html_handlebars/helpers/toc.rs [hunk 1]
```diff
@@ -75,6 +75,7 @@ impl HelperDef for RenderToc {
                 ("", 1)
             };
 
+            #[warn(clippy::if_same_then_else)]
```
Comment (ehuss) [inline src/renderer/html_handlebars/helpers/toc.rs:78]: I'm not sure why this is added.

File: src/theme/index.hbs [hunk 2]
```diff
@@ -1,289 +1,315 @@
...
-        <script type="text/javascript">
-            var html = document.querySelector('html');
-            var sidebar = 'hidden';
-            if (document.body.clientWidth >= 1080) {
-                try { sidebar = localStorage.getItem('mdbook-sidebar'); } catch(e) { }
-                sidebar = sidebar || 'visible';
-            }
-            html.classList.remove('sidebar-visible');
-            html.classList.add("sidebar-" + sidebar);
-        </script>
```
Comment (ehuss) [inline src/theme/index.hbs:3]: I'd prefer to avoid unrelated formatting changes. Can you remove all the whitespace changes and formatting changes?

**Comment by ehuss** [comment]
@Dylan-DPC does this seem like a reasonable idea to you? I suspect there might be some longer term issues with making it easier to extend the templates (#353 ?), but this seems like a simple, incremental improvement.

**Comment by Dylan-DPC** [comment]
i don't see any issue with this

File: src/renderer/html_handlebars/helpers/toc.rs (hunk 1 above)
Comment (GillesRasigade) [reply src/renderer/html_handlebars/helpers/toc.rs:78]: I added this to avoid having a error after executing `clippy`:
Screenshot from 2020-05-10 10-17-35

File: src/theme/index.hbs (hunk 2 above)
Comment (GillesRasigade) [reply src/theme/index.hbs]: Code auto-formatting has been reverted.

**Comment by GillesRasigade** [comment]
Documentation added. Comments answered. Thanks for your reviews.

File: src/renderer/html_handlebars/helpers/toc.rs (hunk 1 above)
Comment (ehuss) [reply src/renderer/html_handlebars/helpers/toc.rs]: Ah, I see.  I just pushed a change to rewrite it: #1214.  I usually prefer unrelated changes like this to be separate.

**Comment by code-ink-space** [comment]
The documentation might be confusing.
https://rust-lang.github.io/mdBook/format/theme/index.html

It says:
> The theme is totally customizable, you can selectively replace every file from the theme by your own by adding a theme directory next to src folder in your project root.

I think it should say "under" or "inside" the src folder because that's what worked for me. Not having `theme` next to `src` but inside it.
Doing `mdbook init --theme` made me see the correct directory structure.
